### PR TITLE
Add TGI monitoring guide through Grafana and Prometheus

### PR DESCRIPTION
**Pull request by fxmarty**
As per title. It is very useful.

### CODE REVIEW

File: docs/source/basic_tutorials/monitoring.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Monitoring TGI server with Prometheus and Grafana dashboard
+
+TGI server deployment can easily be monitored through a Grafana dashboard, consuming a Prometheus data collection. Example of inspectable metrics are statistics on the effective batch sizes used by TGI, prefill/decode latencies, number of generated tokens, etc.
+
+In this tutorial, we look at how to set up a local Grafana dashboard to monitor TGI usage.
+
+![Grafana dashboard for TGI](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/grafana.png)
```
Comment (Narsil) [inline docs/source/basic_tutorials/monitoring.md:7]: I kind of like this graph too:

Screenshot 2024-05-16 at 16 36 20

It's not super useful to detect core issues, but it's more stable metrics (and what people tend to boast within benchmarks)

**Review by Narsil — COMMENTED** [review]
Thanks A LOT for this.

IMHO the screenshot should be very high up in the README page.

**Review by drbh — APPROVED** [review]
thank you! this is super helpful.

nit: I agree a screenshot with more metrics/graphs on the top of the docs would look best

**Comment by fxmarty** [comment]
@Narsil @drbh I lost the approval.

I updated the image and the recommended JSON, it's very nice, thank you! https://moon-ci-docs.huggingface.co/docs/text-generation-inference/pr_1908/en/basic_tutorials/monitoring

**Review by drbh — APPROVED** [review]
looks great! thank @fxmarty!

**Pull request merged**